csrc/scheduler/pointwise_utils.cpp

```diff
@@ -78,10 +164,34 @@ void DomainMap::eraseIfInputMappedThroughRFactorDomain(
     exact_sets.pushBack(ca_map_.disjointSetOf(id, IdMappingMode::EXACT));
   });
 
-  // This technically traverses exprs that are not rfactor
-  // transformations but when this function is used, there should be
-  // no non-rfactor IterDomain exprs
-  auto all_exact_sets_covered = ca_map_.getAllDisjointSetProducers(exact_sets);
+  // Traverse through indexed domains.
+  const auto indexed_sets = getIndexedProducerConsumerPairs(fusion_, ca_map_);
+
+  VectorOfUniqueEntries<std::shared_ptr<VectorOfUniqueEntries<IterDomain*>>>
+      all_exact_sets_covered;
+
+  // Back traverses through the exact map and indexed
+  // producer-consumer pairs
+  for (auto current_sets = exact_sets; !current_sets.empty();) {
+    auto producer_sets = ca_map_.getAllDisjointSetProducers(current_sets);
+    all_exact_sets_covered.pushBack(producer_sets);
```

Comment:
Should we push back the entire `producer_sets`, or should we just push back items where the check below `indexed_it != indexed_sets.end()` goes `false`?

Reply:
I think pushing back the entire sets is fine. The sets in `current_sets` are those that are not yet traversed, so `producer_sets` should also be new sets.

The loop below is to traverse across indexed domains. We need this since there's nothing connecting them. I debated myself whether we should have a new IterDomain expr, or maybe we could use another ID map that allows mapping of such domains. I'm not really sure what would be the best design at this moment, so this function is making this traversal in an ad-hoc way. This may be another place where the new ID map might be helpful.

Reply:
Yeah, I was also thinking about a new ID map. I think it is fine for this PR, in the future, if we see this pattern repeat, then we should add this new ID map type.